packages/forms/src/widgets/CodeWidget/CodeWidget.component.js

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable no-console */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/no-mutable-exports */
+/* eslint-disable global-require */
+
+import React, { PropTypes } from 'react';
+import TextareaWidget from 'react-jsonschema-form/lib/components/widgets/TextareaWidget';
+
+function TextareaCodeWidget(props) {
+	return <TextareaWidget {...props} />;
+}
+TextareaCodeWidget.displayName = 'TextareaCodeWidget';
+
+let CodeWidget = TextareaWidget;
+const SET_OPTIONS = {
+	enableBasicAutocompletion: true,
+	enableLiveAutocompletion: true,
+	enableSnippets: true,
+};
+const DEFAULT_LANGUAGES = ['javascript', 'java', 'python', 'sql', 'json'];
+
+try {
+	const AceEditor = require('react-ace').default;
+
+	try {
+		require('brace/theme/monokai');
+		require('brace/ext/language_tools');
+	} catch (e) {
+		console.log(e);
+	}
+	DEFAULT_LANGUAGES.forEach((lang) => {
+		try {
+			require(`brace/mode/${lang}`);
+			require(`brace/snippets/${lang}`);
+		} catch (e) {
+			console.log(e);
+		}
+	});
+
+	class AceCodeWidget extends React.Component {
+		static displayName = 'AceCodeWidget';
+
+		constructor(props) {
+			super(props);
+			this.onChange = this.onChange.bind(this);
+			this.onLoad = this.onLoad.bind(this);
+		}
+
+		onChange(value) {
+			if (this.props.onChange) {
+				this.props.onChange(value);
+			}
+		}
+
+		onLoad(event) {
+			if (this.props.formContext.codeWidgetOnLoad) {
+				this.props.formContext.codeWidgetOnLoad(event.env.editor);
+			}
+		}
+
+		render() {
+			const {
+				id,
+				options,
+				value,
+				disabled,
+				readonly,
+				autofocus,
+			} = this.props;
+			if (options && options.language && DEFAULT_LANGUAGES.indexOf(options.language) === -1) {
+				console.error(`${options.language} language not supported`);
```

Comment:
this error is fine, but maybe we could add a visual cue when the component render ?